compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -8086,7 +8086,13 @@ public void visit(BLangStatementExpression bLangStatementExpression) {
     public void visit(BLangQueryExpr queryExpr) {
         boolean prevIsVisitingQuery = this.isVisitingQuery;
         this.isVisitingQuery = true;
-        BLangStatementExpression stmtExpr = queryDesugar.desugar(queryExpr, env, getVisibleXMLNSStmts(env));
+        BLangStatementExpression stmtExpr;
+        if (queryExpr.stmtExpr == null) {
+            stmtExpr = queryDesugar.desugar(queryExpr, env, getVisibleXMLNSStmts(env));
+            queryExpr.stmtExpr = stmtExpr;
+        } else {
+            stmtExpr = queryExpr.stmtExpr;
+        }
         result = rewrite(stmtExpr, env);
```

Comment:
Can't we just do
```suggestion
        queryExpr.stmtExpr = queryDesugar.desugar(queryExpr, env, getVisibleXMLNSStmts(env));
        result = rewrite(queryExpr.stmtExpr, env);
```
here?